### PR TITLE
update lesson 5 and add lesson 7 and appendix

### PR DESCRIPTION
**Pull request by piyueh**
This PR:
* in lesson 5:
  * runs the optimization with a fixed number of iterations, record the loss history, and use the parameters at the lowest validation loss as the solution.
  * uses $L$ instead of $\mathrm{loss}$
  * uses the means instead of the sums for both the loss and regularization term
  * decouples the calculations of the raw loss and the regularization 
  * monitors the raw loss instead of the regularization loss
* adds lesson 7
* adds an appendix to explain terms

To see a cell by cell diff of lesson 5: [Screenshot 2021-08-10 at 14-06-38 nbdime - diff and merge your Jupyter notebooks](https://user-images.githubusercontent.com/7220966/128912049-84bf22d6-56fa-48d8-b84e-5add7d1b3d00.png)

### CODE REVIEW

**Comment by piyueh** [comment]
~~check the regularized and the original loss after the training.~~
Done. Turns out I was using different datasets for the raw and regularized losses.
